Moving the solution structure structure piece to designing of messages (#5436)
Particular/docs.particular.net: nservicebus/messaging/messages-events-commands.md

Generic message definitions (e.g. `MyMessage<T>`) are not supported. It is recommended to use dedicated, simple types for each message or to use inheritance to reuse shared message characteristics.

### Solution structure

Messages define the data contract between two endpoints.

It's recommended to use a dedicated assembly for message definitions. By keeping messages in a separate assembly, the amount of information and dependencies shared between services is minimized. It is recomended to have a separate message assembly for every service. Due to that, service can [evolve its contracts](/nservicebus/messaging/evolving-contracts.md) without impacting other services in the system. Every message should be declared in the message assembly of the service owning that message. 

> Consideration should be given to thinking about how the contract assembly will be used by services. When certain events are subscribed by multiple endpoints managed by other teams, it might make sense to publish those contracts as a separate NuGet package. Depending on the use and frequency of changes it could make sense to seprate contracts into multiple assemblies. 

Reviewer: ```suggestion
Consideration should be given to how the contracts assembly will be used by services. When certain events are subscribed to by multiple endpoints managed by other teams, it might make sense to extract those contracts into a separate NuGet package. Depending on the use and the frequency of changes it could make sense to separate contracts into multiple assemblies. 
```

> It's also possible to share messages as C# source files without packaging them into an assembly. One advantage of this approach is that messages don't need to be compiled against specific NServiceBus versions, so assembly redirects are not necessary. This can be accomplished by using [unobstrusive mode messages](/nservicebus/messaging/unobtrusive-mode.md) as well.

Reviewer: ```suggestion
It's also possible to share messages as C# source files without packaging them into an assembly. One advantage of this approach is that messages don't need to be compiled against specific NServiceBus versions, which means that assembly redirects are redundant. This can also be accomplished by making use of the [unobstrusive mode](/nservicebus/messaging/unobtrusive-mode.md).
```

Messages define the data contract between two endpoints.

> It's recommended to use a dedicated assembly for message definitions. By keeping messages in a separate assembly, the amount of information and dependencies shared between services is minimized. It is recomended to have a separate message assembly for every service. Due to that, service can [evolve its contracts](/nservicebus/messaging/evolving-contracts.md) without impacting other services in the system. Every message should be declared in the message assembly of the service owning that message. 

Reviewer: ```suggestion
It's recommended to use a dedicated assembly for message contracts. By keeping message contracts in a separate assembly, the amount of information and dependencies shared between services is minimized. It is recommended to have a separate message assembly for every service. When doing so, a service can [evolve its contracts](/nservicebus/messaging/evolving-contracts.md) without impacting other services in the system. Every message contract should be declared in the contracts assembly of the service owning that message contract. 
```
